src/Http/Routing/src/EndpointRoutingMiddleware.cs

```diff
@@ -102,6 +108,35 @@ private Task SetRoutingAndContinue(HttpContext httpContext)
             }
 
             Log.MatchSuccess(_logger, endpoint);
+
+            var shortCircuitMetadata = endpoint.Metadata.GetMetadata<ShortCircuitMetadata>();
+            if (shortCircuitMetadata is not null)
+            {
+                if (!_routeOptions.SuppressCheckForUnhandledSecurityMetadata)
```

Comment:
This should be shared code so they don't get out of sync.

Reply:
I did try that first, but the conditions and the exception messages are different. In endpoint middleware it would check for the existence of the auth middleware but here that check would be irrelevant since the auth middleware is usually after routing middleware.